### PR TITLE
Implemented SeleneFire

### PR DESCRIPTION
**Pull request by reverienest**

### CODE REVIEW

File: Assets/CharTest/Scripts/HeavyBullet.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        }
+        else
+        {
+            return true;
+        }
+    }
+    public void SpawnChildren(Vector2 newHeavyShot, float newAngle, int heavyProjectiles)
+    {
+        for (int i = 0; i < heavyProjectiles + 1; i++)
+            {
```
Comment (BrandonShockley) [inline Assets/CharTest/Scripts/HeavyBullet.cs]: Fix indentation

Comment (BrandonShockley) [inline Assets/CharTest/Scripts/HeavyBullet.cs:61]: Fix indentation

File: Assets/CharTest/Scripts/HeavyBullet.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    [HideInInspector]
+    public bool hasMutated;
+
+    private Rigidbody2D rb;
+
+    [HideInInspector]
+    public Vector2 TargetDir() { return (target.position - transform.position).normalized;}
+    // Start is called before the first frame update
+
+  void Start()
```
Comment (BrandonShockley) [inline Assets/CharTest/Scripts/HeavyBullet.cs:28]: Fix indentation

File: Assets/Game/Character/Selene/States/SeleneIdleState.cs [hunk 3]
```diff
@@ -16,13 +16,13 @@ override public void Update(SeleneStateInput input)
...
-        if (SpellMap.Instance.GetSpellDown(input.cc.playerNumber, SpellType.INTRIN))
+        if (SpellMap.Instance.GetSpellDown(input.cc.playerNumber, SpellType.MOVE))
         {
             character.ChangeState<SeleneTeleportState>();
         }
-        // TODO: This triggers the last word. Yun can remove this to cast the primary spell.
-        if (SpellMap.Instance.GetSpellDown(input.cc.playerNumber, SpellType.PRIM))
-        {
+        // TODO: Currently maps the Last Word to the proper key, need to confinure with respect to MC gauge. 
+       if (InputMap.Instance.GetInputDown(input.cc.playerNumber, ActionType.LAST_WORD))
```
Comment (BrandonShockley) [inline Assets/Game/Character/Selene/States/SeleneIdleState.cs:24]: The code to activate last word is in SharedCharacterController. Remove this.

File: Assets/CharTest/Scripts/SeleneFire.cs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                bool mutateCheck = bulletList[i].CheckMutation();
+                if (mutateCheck != true)
+                {
+                    StartCoroutine(bulletList[i].IntrinsicMutate());
+                }
+            }   
+        }
+    }   
+
+    IEnumerator SelenePrimary(int _primaryProjectiles, int _primaryWaves, Vector2 startShot)
```
Comment (BrandonShockley) [inline Assets/CharTest/Scripts/SeleneFire.cs:66]: You don't need to pass in primaryProjectiles or primaryWaves when you can access them as member variables

File: Assets/CharTest/Scripts/SeleneFire.cs [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+                angle += 60f / (tempProjectiles + 1);
+
+            }
+            yield return new WaitForSeconds(.4f);
+            tempProjectiles = tempProjectiles - 1; 
+        }
+        
+    }
+
+    IEnumerator SeleneHeavy(int _heavyProjectiles)
```
Comment (BrandonShockley) [inline Assets/CharTest/Scripts/SeleneFire.cs:96]: Likewise here

File: Assets/CharTest/Scripts/SeleneFire.cs [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    {
+        float angle = Mathf.Atan2(TarDir.y, TarDir.x) * Mathf.Rad2Deg;
+        float shotDirXPos = Mathf.Cos(angle * Mathf.Deg2Rad);
+        float shotDirYPos = Mathf.Sin(angle * Mathf.Deg2Rad);
+
+            Vector2 shotDirection = new Vector2(shotDirXPos, shotDirYPos) * primarySpeed;
```
Comment (BrandonShockley) [inline Assets/CharTest/Scripts/SeleneFire.cs:102]: Indentation

File: Assets/CharTest/Scripts/SeleneFire.cs [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    [SerializeField]
+    private int heavyProjectiles;       //Number of split heavy projectiles
+
+    public GameObject primaryPrefab;  
+    public GameObject heavyPrefab;
+
+    private Vector2 startShot;
+    [HideInInspector]
+    public Vector2 TarDir;
+    public CharacterTargeting CharTar;
```
Comment (BrandonShockley) [inline Assets/CharTest/Scripts/SeleneFire.cs:23]: For style conformity, variables names should begin with a lowercase letter

File: Assets/CharTest/Scripts/SeleneFire.cs [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    private int primaryProjectiles;     //Number of projectiles in first wave of primary spell
+    [SerializeField]
+    private int primaryWaves;           //Number of waves in primary spell
+    [SerializeField]
+    private float primarySpeed;         //Speed of projectiles in primary spell
+    [SerializeField]
+    private int heavyProjectiles;       //Number of split heavy projectiles
+
+    public GameObject primaryPrefab;  
+    public GameObject heavyPrefab;
```
Comment (BrandonShockley) [inline Assets/CharTest/Scripts/SeleneFire.cs:18]: These can be
[SerializeField]
private ...

There are other parts of your code that could be changed like this. Have a look around.

File: Assets/CharTest/Scripts/SeleneFire.cs [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+    private Vector2 startShot;
+    [HideInInspector]
+    public Vector2 TarDir;
```
Comment (BrandonShockley) [inline Assets/CharTest/Scripts/SeleneFire.cs:22]: Is there a reason you're making this public?

File: Assets/CharTest/Scripts/SeleneFire.cs [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+            tempProjectiles = tempProjectiles - 1; 
+        }
+        
+    }
+
+    IEnumerator SeleneHeavy(int _heavyProjectiles)
+    {
+        float angle = Mathf.Atan2(TarDir.y, TarDir.x) * Mathf.Rad2Deg;
+        float shotDirXPos = Mathf.Cos(angle * Mathf.Deg2Rad);
+        float shotDirYPos = Mathf.Sin(angle * Mathf.Deg2Rad);
```
Comment (BrandonShockley) [inline Assets/CharTest/Scripts/SeleneFire.cs:100]: This Vector -> Angle -> Vector code is unnecessary as far as I can tell. Just use TarDir as the direction vector.

**Review by BrandonShockley — CHANGES_REQUESTED** [review]

File: Assets/CharTest/Scripts/SeleneFire.cs (hunk 9 above)
Comment (reverienest) [reply Assets/CharTest/Scripts/SeleneFire.cs:22]: It was in a previous iteration where I was testing HeavyBullet

**Pull request closed without merging**